### DISCUSSION TITLE
Make TextCaseStartReplacingCommand more flexible

### DISCUSSION THREAD

**Discussion by peterfication** [johmsalas/text-case.nvim#138] [Ideas]
Right now it only considers the full current word. It would be nice if it would use the part where the cursor is on.

- For `CarList` with the cursor on `Car` it would open `Subs/Car/`
- For `EnhancedCarList` with the cursor on `Car` it would also open `Subs/Car/`

Issue https://github.com/johmsalas/text-case.nvim/issues/135 with PR https://github.com/johmsalas/text-case.nvim/pull/137 tackles the first case. The second case is more complex to implement.

**Comment by peterfication** [comment]
In the end, PR #137 implemented both cases.

**Comment by johmsalas** [comment]
**Dot repeatability**. Just thinking in edge cases. I don't think it should support dot repeating. Curious what you find in your workflow

**Comment by peterfication** [comment]
I think it can be "too big" to be dot repeatable. Normally, I such cases I use it and do adjustments to the file. Then I might make similar changes to another file. But it's ok to write it out again or go back in command history I'd say.
